src/store/protocolDataSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { providers } from 'ethers';
+import {
+  availableMarkets,
+  getNetworkConfig,
+  getProvider,
+  marketsData,
+} from 'src/utils/marketsAndNetworksConfig';
+import { StateCreator } from 'zustand';
+
+import { CustomMarket, MarketDataType } from '../ui-config/marketsConfig';
+import { NetworkConfig } from '../ui-config/networksConfig';
+import { RootStore } from './root';
+import { setQueryParameter } from './utils/queryParams';
+
+export interface ProtocolDataSlice {
+  currentMarket: CustomMarket;
+  currentMarketData: MarketDataType;
+  currentChainId: number;
+  currentNetworkConfig: NetworkConfig;
+  jsonRpcProvider: () => providers.Provider;
+  setCurrentMarket: (market: CustomMarket) => void;
+}
```

Comment:
`ProtocolData` seems too generic of a name for this.  This could be improved in one of two ways:

1. Rename to `MarketNetworkSlice`
2. Split out to two slices `MarketSlice` and `NetworkSlice` (I prefer this over 1)

Reply:
Even though markets and networks have lots in common, they are two different things and could be separated.  Network config, jsonRpcProvider, and chain ID make sense to be in network slice, and the rest in the market slice.

Reply:
Good one, in other projects I've had Web3Slice (I know it's even more generic name) which stored everything related to wallet, RPCs and services wrappers around typechain 
https://github.com/bgd-labs/fe-shared-examples/blob/main/front-end/src/web3/store/web3Slice.ts 
so it's more like a dependency injection slice which is responsible for providing proper instances of anything related to web3 to other slices. So we can just use `get().activeWallet` or `get().dataService.getSomeData()` being sure that everything is connected to proper RPC and signer. 
What do you think? I think it's similar to `NetworkSlice` but I don't think having only network related data would be that useful, so maybe we should introduce `Web3Slice`

Reply:
I kept this name when starting off to be at least breaking as possible with your current code & naming.
`protocolDataSlice` contains only the things previously contained in `protocolDataContext`

Reply:
Good points.  Because the two are so intertwined together and used with each other, I'm onboard with the idea of keeping it as a single slice that may represent all the very common pieces of data that we get from a web3 provider.  So yes, we can just rename this since the previous name was just way too generic, even if it's used in other apps or was used previously.  I think `Web3Slice` works, or maybe better for readability as `Web3DataSlice`, so then it reads as `web3Data.someMethodOrValue`.

Reply:
Looking at comments fresh after the weekend, I think it would be better to have the same structure as context, but move logic to `zustand`, at least for the first PR, so it would be easier to track how things were done before and how to do it with zustand, changing both implementation and the logic behind the implementation would make very frustrating to reason is the change was done because of zustand or because the logical error. 
If we can agree that current PR has no new bugs, and we understand the purpose of introducing zustand, I think it's better to merge and later introduce smaller PRs to focus on each particular feature, i.e. renaming, transactions flow, removing context etc.
The purpose of this PR is again to lay the groundwork for new features and unlock refactoring without huge merge conflicts

Reply:
Agree with @satanworker for now lets first handle the large re architecture and then we can move into refactoring some of the naming so we dont do too much as this PR is already quite large